testbed/datasenders/syslog.go

```diff
@@ -102,13 +103,23 @@ func (f *SyslogWriter) Send(lr pdata.LogRecord) error {
 	sdid := strings.Builder{}
 	sdid.WriteString(fmt.Sprintf("%s=\"%s\" ", "trace_id", lr.TraceID().HexString()))
 	sdid.WriteString(fmt.Sprintf("%s=\"%s\" ", "span_id", lr.SpanID().HexString()))
-	sdid.WriteString(fmt.Sprintf("%s=\"%d\" ", "trace_flags", lr.Flags()))
+	sdid.WriteString(fmt.Sprintf("%s=\"%d\"", "trace_flags", lr.Flags()))
 	lr.Attributes().Range(func(k string, v pdata.AttributeValue) bool {
-		sdid.WriteString(fmt.Sprintf("%s=\"%s\" ", k, v.StringVal()))
+		var str string
+		switch v.Type() {
+		case pdata.AttributeValueSTRING:
+			str = v.StringVal()
+		case pdata.AttributeValueDOUBLE:
+			str = strconv.FormatFloat(v.DoubleVal(), 'g', -1, 64)
+		case pdata.AttributeValueINT:
+			str = strconv.FormatInt(v.IntVal(), 10)
+		case pdata.AttributeValueBOOL:
+			str = strconv.FormatBool(v.BoolVal())
+		}
+		sdid.WriteString(fmt.Sprintf(" %s=\"%s\"", k, str))
 		return true
 	})
-	msg := fmt.Sprintf("<166> %s localhost %s - - [%s] %s\n", ts, lr.Name(), sdid.String(), lr.Body().StringVal())
-
+	msg := fmt.Sprintf("<86>1 %s 192.168.1.1 %s 23108 ID52020 [otel %s] %s\n", ts, lr.Name(), sdid.String(), lr.Body().StringVal())
```

Comment:
What was wrong here that could be the reason for failing test? Was there some specific attribute value that was causing the problem?

Reply:
yep, you are right. 
 if log-collection got wrong data, will throw an error. it makes the server unstable. (because UDP server got a larger request. it's need resource(cpu/mem) to handle

Reply:
we had TCP Syslog receiver perf test. and used wrong data and it's working right(although have parse error log).

so this changes not fix the root problem. but it's make UDP server test more stable